### DISCUSSION TITLE
24.2 RC1 Performance Report

### DISCUSSION THREAD

**Discussion by shocknet-justin** [btcsuite/btcd#2176] [Q&A] [answered]
Not an issue perse, but I wanted to provide the anecdote and offer to gather any additional debugging info that may be helpful  as I recall reading that performance should be improved in this release.

I just got a new piece of hardware to test that out, and have found that initial sync is still quite bad in the latest RC (downloaded binaries).

In it's first 48 hours, this new system has so far only sync'd to block 650k and now validating roughly 1.25 blocks/second

- Intel i7-13700
- 32GB of DDR5
- NVME drive
- Debian 12.5 (fresh, minimal)

addrindex and txindex **are** enabled

CPU utilization oscillates between 15-25% with btcd using 34 threads. RAM utilization is negligible. Internet bandwidth for headers is not a bottleneck.

**Comment by Roasbeef** [comment, marked as answer]
(converted to a discussion) 

> addrindex and txindex are enabled

I think you'll find that dropping the addrindex will give you a dramatic speed up. It's the largest index we maintain on disk, and will cause a lot of extra I/O that the new utxo set cache doesn't help with.

**Reply by shocknet-justin** [reply]
Wow, yea from the jump a 5-10x even with the nvme drive

Was under the impression for some reason this was an optimal flag for neutrino clients? That not/no longer the case?
